app/mailers/host_mailer.rb

```diff
@@ -39,7 +39,8 @@ def summary(options = {})
   def error_state(report)
     report = Report.find(report)
     host = report.host
-    owners = host.owner.recipients_for(:puppet_error_state)
+    owners = host.owner.recipients_for(:puppet_error_state) if host.owner.present?
+    owners ||= []
     raise ::Foreman::Exception.new(N_("unable to find recipients")) if owners.empty?
```

Comment:
nope.
if host.owner is present, it will raise if the recipients_for is an empty array.
if host.owner if empty, it will raise because of line 43 (we set owners to an empty array)
(if line 43 wasn't there, owners would raise "undefined method owners or host mailer" because nothing set the owners local variable)